### PR TITLE
Remove initTooltips include & add upgrade considerations

### LINKED ISSUE #10668: Convert initTooltips to Stimulus
Follow-up to #8565 (and more recently #10620). We no longer use Bootstrap for tooltips. Some of our tooltips were switched over to Tippy.js before the introduction of Stimulus in Wagtail, and it’d be great to convert them over to the new `w-tooltip` controller introduced in #10620.

Specifically, it’s the vanilla JS `initTooltips` function looking for `data-tippy-content` attributes that we need to replace with Stimulus. For example:

```diff
<a href="{% url 'wagtailadmin_pages:history' parent_page.id %}"
    class="{{ nav_icon_button_classes }}"
+   data-controller="w-tooltip"
+   data-w-tooltip-content-value="{% trans 'History' %}"
-    data-tippy-content="{% trans 'History' %}"
    data-tippy-offset="[0, 0]"
+    data-w-tooltip-placement-value="bottom"
-    data-tippy-placement="bottom"
    aria-label="{% trans 'History' %}"
>
    {% icon name="history" classname=nav_icon_classes %}
</a>
```

Files
1. wagtail/admin/templates/wagtailadmin/pages/page_listing_header.html
2. wagtail/admin/templates/wagtailadmin/shared/human_readable_date.html
3. wagtail/admin/templates/wagtailadmin/shared/page_status_tag_new.html
4. wagtail/admin/templates/wagtailadmin/shared/side_panel_toggle.html
5. wagtail/admin/templates/wagtailadmin/shared/headers/page_edit_header.html
6. wagtail/admin/templates/wagtailadmin/shared/workflow_history/list.html
7. wagtail/contrib/modeladmin/tests/test_simple_modeladmin.py (unit test update)
8. wagtail/snippets/templates/wagtailsnippets/snippets/headers/edit_header.html

The main benefit here is simpler initialisation of those tooltip components regardless of where they are used. Additionally we could also convert more of those Tippy.js data-tippy to Stimulus-powered equivalents, though this isn’t needed for what the tooltips are used for currently.

Here is a rough start on how to enhance `TooltipController` for support of the `offset` value.

```diff
// FILE: client/src/controllers/TooltipController.ts
export class TooltipController extends Controller<HTMLElement> {
  static values = {
    content: String,
+    offset: Array,
    placement: { default: 'bottom', type: String },
  };

+  declare readonly hasOffsetValue: boolean;
  declare contentValue: string;
+  declare offsetValue: [number, number];
  declare placementValue: Placement;
  tippy?: Instance<Props>;

  // ...
  get options(): Partial<Props> {
    return {
      content: this.contentValue,
      placement: this.placementValue,
      plugins: [hideTooltipOnEsc],
+      ...(this.hasOffsetValue && { offset: this.offsetValue }),
    };
  }
// .. 
```

Additionally, here is a rough unit test to check that the offset value works correctly.

```js
// FILE client/src/controllers/TooltipController.test.js
  it('should support passing the offset value', async () => {
    document.body.innerHTML = `
    <section>
      <button
        id="button"
        type="button"
        data-controller="w-tooltip"
        data-w-tooltip-offset-value="[10, 20]"
      >
        CONTENT
      </button>
    </section`;

    await Promise.resolve(requestAnimationFrame);

    const tippy = application.getControllerForElementAndIdentifier(
      document.getElementById('button'),
      'w-tooltip',
    ).tippy;

    expect(tippy.props).toHaveProperty('offset', [10, 20]);
  });
```

Once this is done, the data-tippy-offset can be replaced with a Stimulus value in all usage.

```diff
-    data-tippy-offset="[0, 0]"
+    data-w-tooltip-offset="[0, 0]"
```

Stimulus will parse this as a JSON string automatically.

Important: Once done...

1. `initTooltips` export from `client/src/includes/initTooltips.ts` can be fully removed and all usage in other code removed.
2. `client/src/includes/initTooltips.ts` can be renamed to `client/src/utils/tooltips.ts`, and imports of things from this file can be updated to the new path.

### PR DESCRIPTION
**Pull request by lb-**
## Overview

- Fully remove initTooltips util and replace with Controller usage, add release notes and backwards compatible support for data-tippy-* (on initial load only) to be removed fully in a future release.
- Adds more unit tests to the tooltip & dropdown controller code, specifically for the hide on esc behaviour.
- Fixes #10668 (last part of the issue), built on #10869

## Checklist

-   [x] Do the tests still pass?
-   [x] Does the code comply with the style guide?
-   [x] For front-end changes: Did you test on all of Wagtail’s supported environments?
-   [x] For new features: Has the documentation been updated accordingly? **upgrade considerations only**

## How to test this change

- In the code changes, the `plugins` are now moved to the controller files with one being exported (hide on esc) as it's used by other code. These functions are 100% the same.
- A key change however, is that we no longer need to dynamically init tooltips in the chooser modal or when dynamic search results get patched into the DOM.

To validate this, do some testing as follows;

1. Load the pages search `http://localhost:8000/admin/pages/search/?q=breads` and then confirm, on initial load, that the time ago style tooltips show.
2. Now, do a search for a different page, when the HTML gets updated with the new results, confirm these new results have functional tooltips.
3. Now, edit the home page (or any page with a page chooser in use), within the chooser modal, confirm the tooltips work correctly in the initial modal results.
4. When, do a search within the modal to confirm that the tooltips still work.

### CODE REVIEW

**Comment by lb-** [comment]
This is now ready for review and is the last bit of removal for the `data-tippy-*` attribute usage, except for a small support for these attributes on initial load. I am happy to add a mutation observer to support this approach for more use cases if we feel that's needed.

Would love to get your eyes on this either @thibaudcolas or @laymonage  - while #10668 is closed, this PR will do the final clean up from that issue.

**Comment by lb-** [comment]
Adding to 5.2 as this is a key part of finishing off our full migration of the tooltips with a deprecation plan in place.

**Comment by lb-** [comment]
Note to self. Bump upgrade considerations to 5.3

**Comment by lb-** [comment]
Rebased, changed to 5.3

**Comment by lb-** [comment]
Rebased, fixed a typo and changed the deprecated comment to be `@deprecated RemovedInWagtail70`. This is in TooltipController - where we will keep supporting the old data attributes on load for some time.

File: docs/releases/6.0.md [hunk 1]
```diff
@@ -239,3 +239,22 @@ class MyPage(Page):
 ### `filter_queryset` and `get_filtered_queryset` methods no longer return filters
 
 The undocumented internal methods `filter_queryset(queryset)` on `wagtail.admin.views.generic.IndexView`, and `get_filtered_queryset()` on `wagtail.admin.views.reports.ReportView`, now return just the filtered queryset; previously they returned a tuple of `(filters, queryset)`. The filterset instance is always available as the cached property `self.filters`.
+
+### `data-tippy-content` attribute support will be removed
```
Comment (thibaudcolas) [inline docs/releases/6.0.md:243]: Great! Looks very actionable to me.

File: client/src/controllers/DropdownController.ts [hunk 2]
```diff
@@ -104,4 +165,12 @@ export class DropdownController extends Controller<HTMLElement> {
       },
     };
   }
+
+  get plugins() {
+    return [
+      hideTooltipOnBreadcrumbsChange,
+      hideTooltipOnEsc,
+      rotateToggleIcon,
+    ].concat(this.hideOnClickValue ? [hideTooltipOnClickInside] : []);
```
Comment (thibaudcolas) [inline client/src/controllers/DropdownController.ts:174]: Hahaha, the lengths we go to for JS expressions over statements!

Comment (lb-) [reply client/src/controllers/DropdownController.ts:174]: Yeah. True, it's probably a bit esoteric to write this way.

File: client/src/controllers/TooltipController.test.js [hunk 3]
```diff
@@ -67,6 +67,29 @@ describe('TooltipController', () => {
...
+    document
+      .querySelector('section')
+      .dispatchEvent(
+        new KeyboardEvent('keydown', { bubbles: true, key: 'Escape' }),
+      );
+
+    await Promise.resolve();
+
+    expect(document.querySelectorAll('[role="tooltip"]')).toHaveLength(0);
+  });
```
[the remark this answers is not in the document]
Comment (lb-) [reply client/src/controllers/TooltipController.test.js:91]: Thanks. Took a bit to get these working but happy we have a baseline of unit tests here. If (or not 🤞) we find bugs, it's great to be able to reinforce is JS fixes with tests.

**Pull request merged**